readium/navigator/src/main/java/org/readium/r2/navigator/HyperlinkNavigator.kt

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2023 Readium Foundation. All rights reserved.
+ * Use of this source code is governed by the BSD-style license
+ * available in the top-level LICENSE file of the project.
+ */
+
+package org.readium.r2.navigator
+
+import org.readium.r2.shared.ExperimentalReadiumApi
+import org.readium.r2.shared.publication.Link
+import org.readium.r2.shared.util.AbsoluteUrl
+
+/**
+ * A navigator supporting hyperlinks.
+ */
+@ExperimentalReadiumApi
+public interface HyperlinkNavigator : Navigator {
+
+    @ExperimentalReadiumApi
+    public interface Listener : Navigator.Listener {
+
+        /**
+         * Called when a link to an internal resource was clicked in the navigator.
+         *
+         * You can use this callback to perform custom navigation like opening a new window
+         * or other operations.
+         *
+         * By returning false the navigator wont try to open the link itself and it is up
+         * to the calling app to decide how to display the link.
+         */
+        @ExperimentalReadiumApi
+        public fun shouldJumpToLink(link: Link): Boolean { return true }
```

Comment:
None of these suggestions imply that we're talking about internal links and navigation. On the contrary, I feel like `jump` implies that we won't leave the navigator, so external links don't trigger that method.

Reply:
My first comment was more of a general conversation and not completely related to `shouldJumpToLink`.

But my preference is for `shouldFollowLink()` here, because `shouldJumpToLink()` doesn't really convey that the move was triggered by a click on a link IMHO. Mainly because of the confusion between `shouldJumpTo**Link**` which is the `<a>` link in the document and the `Link` object which is part of the reading order.

Reply:
We settled on `shouldFollowInternalLink`.